### ISSUE TITLE
typescript-resolvers: maximum call stack size exceeded

### ISSUE THREAD

**Issue by yoavain-sundaysky** [dotansimha/graphql-code-generator#2030]
**Describe the bug**
typescript-resolvers is getting "maximum call stack size exceeded" error when there are 2 types pointing at each other.

Example:
```graphql
type Account {
	id: ID!
	name: String!
	programs: [Program!]!
}

type Program {
	id: ID!
	name: String!
	account: Account!
}
```

3. My `codegen.yml` config file:

```yml
overwrite: true
schema:
  - ./src/graphql/**/*.graphql
generates:
  ./generated/graphqlGeneratedTypes/graphqlServer.ts:
    config:
      typesPrefix: "Gql"
      defaultMapper: Partial<{T}>
      namingConvention:
        typeNames: change-case#pascalCase
        enumValues: change-case#upperCase
    plugins:
      - typescript
      - typescript-resolvers
```

- OS: Windwos
- `@graphql-codegen/1.3.0`
- NodeJS: 10.x

Worked with 1.2.1

**Comment by mehmetnyarar** [comment]
I receive the same error.

@yoavain-sundaysky can you try removing `typescript-resolvers`? It works fine when I do but of course the output has no resolver types which breaks my code.

**Comment by yoavain-sundaysky** [comment]
the problem seems to be in the typescript-resolvers plugin.
Typescript plugin works fine on my entire project.
I cut my schema to the smallest part that reproduces the problem.
Seems the problem is in the cyclic reference (A -> B, B ->A)

**Comment by dotansimha** [comment]
Thanks @yoavain-sundaysky , I'm taking a look now. We did changed some things related to `typescript-resolvers`.  
As a workaround, you can keep using 1.2.1

**Comment by dotansimha** [comment]
@yoavain-sundaysky I think I managed to fix that. Can you please try `1.3.1-alpha-671765f6.2` and let me know if it works?

**Comment by yoavain-sundaysky** [comment]
1.3.1-alpha-671765f6.2 works. Thanks!

**Comment by chill5018** [comment]
1.3.1-alpha-671765f6.2 works for me too! Thanks  @dotansimha !

**Comment by dotansimha** [comment]
Fixed in 1.3.1
